### PR TITLE
[collector] fix: move 'relabelings' and 'metricRelabelings' configs under endpoints in serviceMonitor spec

### PR DESCRIPTION
**Pull request by rl0nergan**
Fixes indentation issues causing an invalid ServiceMonitor spec when using relabeling/metricRelabeling configs  https://github.com/open-telemetry/opentelemetry-helm-charts/issues/1287

### CODE REVIEW

File: charts/opentelemetry-collector/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -17,11 +17,11 @@ spec:
   endpoints:
   {{- toYaml .Values.serviceMonitor.metricsEndpoints | nindent 2 }}
   {{- with .Values.serviceMonitor.relabelings }}
```
Comment (jaronoff97) [inline charts/opentelemetry-collector/templates/servicemonitor.yaml:19]:
```suggestion
    {{- with .Values.serviceMonitor.relabelings }}
```
To match with below

Comment (rl0nergan) [reply charts/opentelemetry-collector/templates/servicemonitor.yaml]: good call. just pushed those changes up

**Review by Allex1 — APPROVED** [review]

**Review by TylerHelmuth — COMMENTED** [review]
@rl0nergan can you do a `helm template` locally with these changes and post the result

**Comment by rl0nergan** [comment]
> @rl0nergan can you do a `helm template` locally with these changes and post the result

yep. using the values below for the serviceMonitor:
```
  serviceMonitor:
    enabled: true
    # additional labels on the ServiceMonitor
    extraLabels: {}
    # add annotations on the ServiceMonitor
    annotations: {}
    metricsEndpoints:
      - port: metrics
    # Used to set relabeling and metricRelabeling configs on the ServiceMonitor
    # https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config
    relabelings:
      - action: replace
        replacement: my-cluster
        targetLabel: cluster
      - action: replace
        replacement: my-cluster2
        targetLabel: cluster2
    metricRelabelings:
      - action: keep
        regex: otelcol_process_memory_rss|target_info
        sourceLabels:
        - __name__
```

Templates out to this:
```
# Source: opentelemetry-collector/templates/servicemonitor.yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: otel-collector-opentelemetry-collector
  namespace: redpanda
  labels:
    helm.sh/chart: opentelemetry-collector-0.100.1
    app.kubernetes.io/name: opentelemetry-collector
    app.kubernetes.io/instance: otel-collector
    app.kubernetes.io/version: "0.105.0"
    app.kubernetes.io/managed-by: Helm
    
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: opentelemetry-collector
      app.kubernetes.io/instance: otel-collector
      component: statefulset-collector
  endpoints:
  - port: metrics
    relabelings:
      - action: replace
        replacement: my-cluster
        targetLabel: cluster
      - action: replace
        replacement: my-cluster2
        targetLabel: cluster2
    metricRelabelings:
      - action: keep
        regex: otelcol_process_memory_rss|target_info
        sourceLabels:
        - __name__
```

**Comment by TylerHelmuth** [comment]
Lets add more indentation so that 

```yaml
  endpoints:
  - port: metrics
    relabelings:
      - action: replace
        replacement: my-cluster
        targetLabel: cluster
      - action: replace
        replacement: my-cluster2
        targetLabel: cluster2
    metricRelabelings:
      - action: keep
        regex: otelcol_process_memory_rss|target_info
        sourceLabels:
        - __name__
```

Become

```yaml
  endpoints:
    - port: metrics
      relabelings:
        - action: replace
          replacement: my-cluster
          targetLabel: cluster
        - action: replace
          replacement: my-cluster2
          targetLabel: cluster2
      metricRelabelings:
        - action: keep
          regex: otelcol_process_memory_rss|target_info
          sourceLabels:
          - __name__
```

**Comment by rl0nergan** [comment]
Changes pushed. Renders like this now:
```
# Source: opentelemetry-collector/templates/servicemonitor.yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: otel-collector-opentelemetry-collector
  namespace: redpanda
  labels:
    helm.sh/chart: opentelemetry-collector-0.101.1
    app.kubernetes.io/name: opentelemetry-collector
    app.kubernetes.io/instance: otel-collector
    app.kubernetes.io/version: "0.106.1"
    app.kubernetes.io/managed-by: Helm
    
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: opentelemetry-collector
      app.kubernetes.io/instance: otel-collector
      component: statefulset-collector
  endpoints:
    - port: metrics
      relabelings:
        - action: replace
          replacement: my-cluster
          targetLabel: cluster
        - action: replace
          replacement: my-cluster2
          targetLabel: cluster2
      metricRelabelings:
        - action: keep
          regex: otelcol_process_memory_rss|target_info
          sourceLabels:
          - __name__
```

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**